### PR TITLE
feat(API): zod validation

### PR DESCRIPTION
**Pull request by jakeaturner**

### CODE REVIEW

**Comment by jakeaturner** [comment]
Current dependencies on/for this PR:
* master
  * **PR #128** <a href="https://app.graphite.dev/github/pr/LibreTexts/conductor/128" target="_blank">Graphite</a> 
    * **PR #129** <a href="https://app.graphite.dev/github/pr/LibreTexts/conductor/129" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/LibreTexts/conductor/129?utm_source=stack-comment).

File: server/api.js [hunk 1]
```diff
@@ -550,17 +541,15 @@ router.route('/commons/catalogs/addresource').put(
   authAPI.verifyRequest,
   authAPI.getUserAttributes,
   authAPI.checkHasRoleMiddleware(process.env.ORG_ID, 'campusadmin'),
-  booksAPI.validate('addBookToCustomCatalog'),
-  middleware.checkValidationErrors,
+  middleware.validateZod(booksAPI.getWithBookIDSchema),
```
Comment (ethanaturner) [inline server/api.js]: `.getWithBookIDBodySchema`

File: server/api.js [hunk 2]
```diff
@@ -550,17 +541,15 @@ router.route('/commons/catalogs/addresource').put(
...
   booksAPI.addBookToCustomCatalog,
 );
 
 router.route('/commons/catalogs/removeresource').put(
   authAPI.verifyRequest,
   authAPI.getUserAttributes,
   authAPI.checkHasRoleMiddleware(process.env.ORG_ID, 'campusadmin'),
-  booksAPI.validate('removeBookFromCustomCatalog'),
-  middleware.checkValidationErrors,
+  middleware.validateZod(booksAPI.getWithBookIDSchema),
```
Comment (ethanaturner) [inline server/api.js]: `.getWithBookIDBodySchema`

**Review by ethanaturner — APPROVED** [review]
_ET_. 10/24/23.

**Pull request merged**